crates/primitives/src/transaction/mod.rs

```diff
@@ -1031,11 +1170,16 @@ impl<'a> arbitrary::Arbitrary<'a> for TransactionSigned {
             transaction.set_chain_id(chain_id % (u64::MAX / 2 - 36));
         }
 
-        let mut tx = TransactionSigned {
-            hash: Default::default(),
-            signature: Signature::arbitrary(u)?,
-            transaction,
+        let signature = Signature::arbitrary(u)?;
+
+        #[cfg(feature = "optimism")]
+        let signature = if transaction.is_deposit() {
+            Signature { r: U256::ZERO, s: U256::ZERO, odd_y_parity: false }
+        } else {
+            signature
         };
+
+        let mut tx = TransactionSigned { hash: Default::default(), signature, transaction };
         tx.hash = tx.recalculate_hash();
```

Comment:
nit: use `TransactionSigned::from_transaction_and_signature`

Reply:
Addressed in https://github.com/anton-rs/op-reth/pull/105